btree/btree.c

```diff
@@ -12391,6 +12420,60 @@ static void ut_lru_test(void)
 	M0_LOG(M0_INFO, "Mem After Free (%"PRId64") || Mem freed (%"PRId64").\n",
 	       mem_after_free, mem_freed);
 
+cleanup:
+	/**
+	 * The test assumes that "limit" will be more than the total number of
+	 * nodes present in the record. Hence only one function call to
+	 * m0_btree_truncate is sufficient.
+	 */
+	cred = M0_BE_TX_CREDIT(0, 0);
+	m0_btree_truncate_credit(tx, tree, &cred, &limit);
+	m0_be_ut_tx_init(tx, ut_be);
+	m0_be_tx_prep(tx, &cred);
+	rc = m0_be_tx_open_sync(tx);
+	M0_ASSERT(rc == 0);
+
+	rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+				m0_btree_truncate(tree, limit, tx,
+							&kv_op));
+	m0_be_tx_close_sync(tx);
+	m0_be_tx_fini(tx);
+
+	/* Verify the tree is empty */
+	M0_ASSERT(m0_btree_is_empty(tree));
+
+	cred = M0_BE_TX_CREDIT(0, 0);
+	m0_be_allocator_credit(NULL, M0_BAO_FREE_ALIGNED, rnode_sz,
+			rnode_sz_shift, &cred);
+	m0_btree_destroy_credit(tree, NULL, &cred, 1);
+
+	m0_be_ut_tx_init(tx, ut_be);
+	m0_be_tx_prep(tx, &cred);
+	rc = m0_be_tx_open_sync(tx);
+	M0_ASSERT(rc == 0);
+
+	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_destroy(tree, &b_op, tx));
+	M0_ASSERT(rc == 0);
+	M0_SET0(&btree);
+
+	/* Delete temp node space which was used as root node for the tree. */
+	buf = M0_BUF_INIT(rnode_sz, rnode);
+	M0_BE_FREE_ALIGN_BUF_SYNC(&buf, rnode_sz_shift, seg, tx);
+
+	m0_be_tx_close_sync(tx);
+	m0_be_tx_fini(tx);
+
+	if (restart_test) {
+		/* When restarting test, Re-map the BE segment. */
```

Comment:
Reset restart_test within this if-condition.

Reply:
Done.